python/setup.py

```diff
@@ -2,7 +2,6 @@
 
 import os
 import skbuild
```

Comment:
About the commit message: I thought we concluded that skbuild only brought setuptoools.command.test into the symbol table by importing it and never overwrote it?

Reply:
We concluded that it now doesn't override setuptools, but they before supplied their own custom test command.
https://github.com/scikit-build/scikit-build/blob/ff9cde427e2ed8f6f683139a23286c063d510352/skbuild/command/test.py

Reply:
Yes, but they did not overwrite setuptoools.command.test in the symbol table, so you would still access setuptools test command when referring directly to setuptools.

Reply:
Yes, agree.
But which exact wording in the commit message you don't like then?

Reply:
"(It is though unclear if test command from setuptools was ever called by
any parts of our solution.)" and "However setuptools.command.test was
imported directly in the skbuild, where they exchanged 'test' with their
own command. So we happened to use it indirectly."

Reply:
1. "(It is though unclear if test command from setuptools was ever called by
any parts of our solution.)" 

It was imported, but was it ever called? Did we just overrode this stuff because we didn't like it and was it ever part of our test execution?
Because I would assume that if us calling test either through skbuild or setuptools was somehow important, then something would have broke down once we removed the override line because some code that is the essence of the setuptools..test method would never be called.

2. I don't understand what is wrong
- setuptools.command test was imported directly in the skbuild by using from setuptools.command.test import test as _test
- it was done in place where they exchanged test with their own command - aka in the skbuild.command.test file
- we happened to use that import (of setuptools.command.test.test) indirectly because we never imported it, yet skbuild did.

Reply:
1. Yes, I do think the setuptools test command was used when running our tests. I think skbuild test command was changed so that It behaves the same as the setuptools test command.

2. I don't think they exchange the setuptools test command. They just define their own. I may have misunderstood your wording here though. I read it as "we were using skbuild test command indirectly", but I think what you ment is "we use the import statement indirectly"?

Reply:
1. Setuptools.test is deprecated, that's why I doubt it is used any longer. And I am not sure it was used because deprecation issue is from 2019.  Because of this uncertainty I added this comment as I feel it is important note.
2. Yes, by exchanged I didn't mean "go inside their code and substitute theirs with custom", but "we have extended their version and exchanged the original call with ours through `kw["cmdclass"]`, so that whenever something in setuptools calls test, our test will be called".
And yes, I mean that we use import statement indirectly.


I suggest you to propose your own wording here, because otherwise we will spend another 10 comments with me trying to figure out how to word that so it pleases you :smile:

Reply:
I can not see any reason why setuptools test command would not be called, so I think it was.

I don't think it is correct to say that anything in setuptools is replaced, rather that skbuild is used in place of setuptools (the setup() function from skbuild is called).

You can have a look at the wording in my PR to segyio (https://github.com/equinor/segyio/pull/576) and see if you agree with it.